app/leaderboard/page.tsx

```diff
@@ -94,18 +96,21 @@ export default function Page() {
     };
 
     setLoading(true);
+    setRankingdataloading(true);
```

Comment:
now if you look closely then adding this new state is similar to using `setLoading` since you use both of them in the exact same places so i dont get why they are both being used. 
What i would suggest is use a function called `fetchPageData` and use like the below - 

```
const fetchPageData=useCallback(async ()=>{
 const requestBody = {
      addr:
        status === "connected"
          ? hexToDecimal(address && address?.length > 0 ? address : userAddress)
          : "",
      page_size: 10,
      shift: 0,
      duration: timeFrameMap(duration),
    };
    setLoading(true);
    await fetchLeaderboardToppersResult({
      addr: requestBody.addr,
      duration: timeFrameMap(duration),
    });
    await fetchRankingResults(requestBody);
    setLoading(false);
},[fetchRankingResults,fetchLeaderboardToppersResult,address,userAddress,status]);
```

and the effect would look something like -

```
useEffect(()=>{
    if (!apiCallDelay) return;
   fetchPageData()
},[apiCallDelay]);
```

Let me know if this solves the problem for you

Reply:
i didn't use the loading/setloading variable because the whole page was rendered based on the loading variable  here `   <div className={styles.leaderboard_container}>
      {loading ? (
        <div className={styles.leaderboard_skeleton}>
          <LeaderboardSkeleton />
        </div>
      ) : (
        <>`
        
       this makes it such that a cause to refetch rank data/leaderbord data will make the page go blank 
       ps: the bove method solved the lag 👍